packages/plugin/src/Integrations/CRM/ActiveCampaign/BaseActiveCampaignIntegration.php

```diff
@@ -0,0 +1,572 @@
+        instructions: 'Enter the name or ID of the desired Stage.',
+        order: 4,
+    )]
+    protected string $stage = '';
+
+    #[Flag(self::FLAG_GLOBAL_PROPERTY)]
+    #[Input\Text(
+        label: 'Owner (Optional if auto-assign enabled in AC)',
+        instructions: "Enter the username or ID of the user you wish to assign as the deal owner. If you don't specify an owner, Active Campaign will auto-assign an owner if it is set up.",
+        order: 5,
+    )]
+    protected string $owner = '';
+
+    public function checkConnection(Client $client): bool
+    {
+        try {
+            $response = $client->get($this->getEndpoint('/webhooks'));
+
+            return 200 === $response->getStatusCode();
+        } catch (\Exception $exception) {
+            throw new IntegrationException($exception->getMessage(), $exception->getCode(), $exception->getPrevious());
+        }
+    }
+
+    public function getApiToken(): string
+    {
+        return $this->getProcessedValue($this->apiToken);
+    }
+
+    public function getPipelineId(): int
+    {
+        return $this->pipelineId;
+    }
+
+    public function setPipelineId(int $pipelineId): self
+    {
+        $this->pipelineId = $pipelineId;
+
+        return $this;
+    }
+
+    public function getStageId(): int
+    {
+        return $this->stageId;
+    }
+
+    public function setStageId(int $stageId): self
+    {
+        $this->stageId = $stageId;
+
+        return $this;
+    }
+
+    public function getOwnerId(): int
+    {
+        return $this->ownerId;
+    }
+
+    public function setOwnerId(int $ownerId): self
+    {
+        $this->ownerId = $ownerId;
+
+        return $this;
+    }
+
+    public function onBeforeSave(Client $client): void
+    {
+        $apiToken = $this->getApiToken();
+        $apiUrl = $this->getApiRootUrl();
+        $pipeline = $this->getPipeline();
+        $stage = $this->getStage();
+        $owner = $this->getOwner();
+
+        // If one of these isn't present, we just return void
+        if (!$apiToken || !$apiUrl) {
+            return;
+        }
+
+        $pipelineId = $this->fetchPipelineId($client, $pipeline);
+        if ($pipelineId) {
+            $this->setPipelineId($pipelineId);
+        }
+
+        $stageId = $this->fetchStageId($client, $stage, $pipelineId);
+        if ($stageId) {
+            $this->setStageId($stageId);
+        }
+
+        $ownerId = $this->fetchOwnerId($client, $owner);
+        if ($ownerId) {
+            $this->setOwnerId($ownerId);
+        }
+    }
+
+    public function fetchFields(string $category, Client $client): array
+    {
+        if ('Contact' === $category) {
+            return $this->fetchContactFields($client);
+        }
+
+        if ('Deal' === $category) {
+            return $this->fetchDealFields($client);
+        }
+
+        if ('Account' === $category) {
+            return $this->fetchAccountFields($client);
+        }
+
+        return [];
+    }
```

Comment:
Lets use the new PHP8.0 `match` for such switch cases.
Also please avoid magic strings and use constants instead.

```suggestion
        match ($category) {
            self::CATEGORY_CONTACT => $this->fetchContactFields($client),
            self::CATEGORY_DEAL => $this->fetchDealFields($client),
            self::CATEGORY_ACCOUNT => $this->fetchAccountFields($client),
            default => [],
        }
    }
```